core/src/main/java/hudson/lifecycle/WindowsServiceLifecycle.java

```diff
@@ -120,6 +120,11 @@ public void rewriteHudsonWar(File by) throws IOException {
 
     @Override
     public void restart() throws IOException, InterruptedException {
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            jenkins.cleanUp();
```

Comment:
I would recommend to catch <code>Exception</code> and to log results as a SEVERE error in Jenkins logs